src/lib.ml

```diff
@@ -632,26 +653,79 @@ end = struct
         Ok ()
       else begin
         visited := Id.Set.add !visited t.unique_id;
-        t.requires >>= fun deps ->
-        Result.List.iter deps ~f:loop >>| fun () ->
+        let* deps = t.requires in
+        let+ () = Result.List.iter deps ~f:loop in
         res := t :: !res
       end
     in
-    Result.List.iter ts ~f:loop >>| fun () ->
+    let+ () = Result.List.iter ts ~f:loop in
     List.rev !res
 
   let associate closure ~orig_stack ~linking =
-    Table.Partial.make closure ~orig_stack
-    >>= fun impls ->
+    let* impls = Table.Partial.make closure ~orig_stack in
     let closure = List.map closure ~f:fst in
     if linking && not (Table.Partial.is_empty impls) then
-      Table.make impls ~orig_stack
-      >>= fun impls ->
+      let* impls = Table.make impls ~orig_stack in
       second_step_closure closure impls
     else
       Ok closure
 end
 
+module Vlib_status = struct
+  type nonrec t =
+    | No_implementation
+    | Implemented_by of t
+    | Too_many_impl of t list
```

Comment:
Why do we need the arguments to the `Implemented_by` and `Too_many_impl` constructors? I don't see them being used anywhere.

Reply:
Indeed they are not useful. They were built for potential error messages but we can remove that actually.